pkg/node/volumemgr.go

```diff
@@ -1213,3 +1215,38 @@ func (m *VolumeManager) addVolumeStatusAnnotation(drive *drivecrd.Drive, volumeN
 	}
 	drive.Annotations[annotationKey] = status
 }
+
+// handleExpandingStatus handles volume CR with Expanding status, it calls ExpandLV to expand volume
+// To get logical volume name it use LVM provisioner function GetVolumePath
+// Receive context, volume CR
+// Return ctrl.Result, error
+func (m *VolumeManager) handleExpandingStatus(ctx context.Context, volume *volumecrd.Volume) (ctrl.Result, error) {
+	ll := m.log.WithFields(logrus.Fields{
+		"method": "handleExpandingStatus",
+	})
+	volumePath, err := m.provisioners[p.LVMBasedVolumeType].GetVolumePath(volume.Spec)
+	if err != nil {
+		ll.Errorf("Failed to get volume path, err: %v", err)
+		return ctrl.Result{Requeue: true}, err
+	}
+	mbSize, err := util.ToSizeUnit(volume.Spec.Size, util.BYTE, util.MBYTE)
+	if err != nil {
+		ll.Errorf("Failed to convert %d to %v, err: %v", volume.Spec.Size, util.MBYTE, err)
+		return ctrl.Result{Requeue: true}, err
+	}
+	if err = m.lvmOps.ExpandLV(volumePath, mbSize); err != nil {
```

Comment:
Is ExpandLV operation idempotent? If so, I think we can try to replace `UpdateCRWithAttempts` with `UpdateCR` function below and try to rely on  `ctrl.Result{Requeue: true}`.